pyradiosky/skymodel.py

```diff
@@ -639,6 +662,201 @@ def __eq__(self, other, check_extra=True):
+
+        No conversion is applied if stokes is already compatible with Jy
+        (for point component_type) or Jy/sr (for healpix component_type).
+        """
+        this_unit = self.stokes.unit
+        if self.component_type == "point":
+            if this_unit.is_equivalent("Jy"):
+                return
+
+        else:
+            if this_unit.is_equivalent("Jy/sr"):
+                return
+
+        if self.spectral_type == "spectral_index" or (
+            self.spectral_type == "flat" and self.reference_frequency is not None
+        ):
+            conv_factor = 1 / skyutils.jy_to_ksr(self.reference_frequency)
+            conv_factor = np.repeat(
+                np.repeat(conv_factor[np.newaxis, np.newaxis, :], 4, axis=0),
+                self.Nfreqs,
+                axis=1,
+            )
+        elif self.freq_array is not None:
+            conv_factor = 1 / skyutils.jy_to_ksr(self.freq_array)
+            conv_factor = np.repeat(
+                np.repeat(conv_factor[np.newaxis, :, np.newaxis], 4, axis=0),
+                self.Ncomponents,
+                axis=2,
+            )
+        else:
+            raise ValueError(
+                "Either reference_frequency or freq_array must be set to convert to Jy."
+            )
+
+        self.stokes = self.stokes * conv_factor
+        self.coherency_radec = skyutils.stokes_to_coherency(self.stokes)
+
+    def jansky_to_kelvin(self):
+        """
+        Apply a conversion to stokes from Jy-based units to K-based units.
+
+        No conversion is applied if stokes is already compatible with K sr
+        (for point component_type) or K (for healpix component_type).
+        """
+        this_unit = self.stokes.unit
+        if self.component_type == "point":
+            if this_unit.is_equivalent("K sr"):
+                return
+
+        else:
+            if this_unit.is_equivalent("K"):
+                return
+
+        if self.spectral_type == "spectral_index" or (
+            self.spectral_type == "flat" and self.reference_frequency is not None
+        ):
+            conv_factor = skyutils.jy_to_ksr(self.reference_frequency)
+            conv_factor = np.repeat(
+                np.repeat(conv_factor[np.newaxis, np.newaxis, :], 4, axis=0),
+                self.Nfreqs,
+                axis=1,
+            )
+        elif self.freq_array is not None:
+            conv_factor = skyutils.jy_to_ksr(self.freq_array)
+            conv_factor = np.repeat(
+                np.repeat(conv_factor[np.newaxis, :, np.newaxis], 4, axis=0),
+                self.Ncomponents,
+                axis=2,
+            )
+        else:
+            raise ValueError(
+                "Either reference_frequency or freq_array must be set to convert to K."
+            )
+
+        self.stokes = self.stokes * conv_factor
+        self.coherency_radec = skyutils.stokes_to_coherency(self.stokes)
+
+    def healpix_to_point(
```

Comment:
I can go either way on this. I added it here because I thought we'd said we wanted such a method, but I'm fine with moving the functionality to pyuvsim.

Reply:
Thinking about this more, I agree that it makes sense to keep it here. We can, if necessary, add more ways of representing diffuse pixels as point sources. The docstring should probably mention that this is treating diffuse pixels as unresolved point sources by integrating over the pixel area.

Reply:
sure, I'll add to the docstring.